src/Container.php

```diff
@@ -228,4 +219,49 @@ private function buildExceptionReport(string $reason): string
 
         return $report . $reason;
     }
+
+    private function throwNonResolvableParameterTypeException(
+        string $qn,
+        ReflectionParameter $param,
+        ?string $reason = null,
+    ): void {
+        $msg = sprintf(
+            'Unable to resolve %s constructor parameter $%s of type %s (position %d).',
+            $qn,
+            $param->getName(),
+            $param->getType() ? (string) $param->getType() : 'unknown',
+            $param->getPosition() + 1,
+        );
+
+        if ($reason !== null) {
+            $msg .= " Reason: $reason";
+        }
+
+        throw new ContainerException($this->buildExceptionReport($msg));
+    }
+
+    private function throwNonResolvableUnionParameterTypeException(
```

Comment:
We can replace this with `throwNonResolvableParameterTypeException()` call, no need for 2 throw methods